src/customnodesession.hpp

```diff
@@ -0,0 +1,57 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "blobmap.hpp"
+#include "custom_node_interface.hpp"
+#include "nodesession.hpp"
+#include "pipelineeventqueue.hpp"
+#include "status.hpp"
+
+namespace ovms {
+
+class ModelManager;
+class Node;
+class NodeLibrary;
+
+class CustomNodeSession : public NodeSession {
+    const NodeLibrary& library;
```

Comment:
Why library is both here & as argument to execute?

Reply:
Nice catch, it is not needed anymore as session field. It was required at some point due to custom node output fetching in ::fetchResults. After moving it to ::execute, having a field is not neccessary.